File: c#/leetcode/739-daily-temperatures.cs
public class Solution {
    public int[] DailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.Length];
        Stack<int[]> stack = new Stack<int[]>();
        
        for (int i=0; i < temperatures.Length; i++) {
            while (stack.Count > 0 && temperatures[i] > stack.Peek()[0]) {
                res[stack.Peek()[1]] = i - stack.Peek()[1];
                stack.Pop();
            }
            stack.Push(new int[]{temperatures[i],i});
        }
        return res;
    }
    
}

File: python/leetcode/739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:

        stack = []
        res = [0] * len(temperatures)

        for i in range(len(temperatures)):
            if not stack or temperatures[i] < stack[-1][0]:
                stack.append((temperatures[i], i))
            else:
                while stack and stack[-1][0] < temperatures[i]:
                    t, idx = stack.pop()
                    res[idx] = i-idx
                stack.append((temperatures[i], i))

        return res
